README.md

```diff
@@ -30,6 +30,45 @@ const diagram = new Appmap.ComponentDiagram('#component-diagram', {
 diagram.render(componentDiagramModel);
 ```
 
+If you want to create your custom context menu for diagram nodes, pass `contextMenu` option with builder function:
+
+<pre>
+const diagram = new Appmap.ComponentDiagram('#component-diagram', {
+  <b>contextMenu: function(componentDiagram){
+    return [
+      (item) => item
+        .text('Set as root')
+        .selector('.nodes .node')
+        .transform((e) => e.getAttribute('id'))
+        .on('execute', (id) => componentDiagram.makeRoot(id)),
+      (item) => item
+        .text('Expand')
+        .selector('g.node')
+        .transform((e) => e.getAttribute('id'))
+        .condition((id) => componentDiagram.hasPackage(id))
+        .on('execute', (id) => componentDiagram.expand(id)),
+      (item) => item
+        .text('Collapse')
+        .selector('g.node')
+        .transform((e) => e.getAttribute('id'))
+        .condition((id) => !componentDiagram.hasPackage(id))
+        .on('execute', (id) => componentDiagram.collapse(id)),
+      (item) => item
+        .text('Reset view')
+        .on('execute', () => {
+          componentDiagram.render(componentDiagram.initialModel);
+        }),
```

Comment:
This API feels a little off standard to me. The menu items were meant to be chained in a similar fashion to the d3 API. Something like this:
```js
diagram.contextMenu((menu) => {
  menu
    .add(item => item.text('Hello').on('execute', () => console.log('Hello world!')))
    .add(item => item.text('Goodbye').on('execute', () => console.log('Goodbye!')));
});
```

Perhaps enabling/disabling default functionality via boolean
```js
diagram.contextMenu(false);
diagram.contextMenu(true);
```

Just thinking out loud and wanted to capture my thoughts. I think this is fine for now.